File: FusionAndroid/app/src/main/java/fr/intechinfo/fusionandroid/CallSafeExecutor.java
package fr.intechinfo.fusionandroid;

import java.io.IOException;

import retrofit2.Call;
import retrofit2.Response;

public class CallSafeExecutor {
    private final Call call;
    public CallSafeExecutor(Call call){
        this.call = call;
    }
    public Response execute() throws Exception {
        try {
            return this.call.execute();
        } catch (IOException e) {
            throw new Exception(e);
        }
    }
}

File: FusionAndroid/app/src/main/java/fr/intechinfo/fusionandroid/CallSafeExecutor.kt
package fr.intechinfo.fusionandroid

import java.io.IOException

import retrofit2.Call
import retrofit2.Response

class CallSafeExecutor(private val call: Call<*>) {
    @Throws(Exception::class)
    fun execute(): Response<*> {
        try {
            return this.call.execute()
        } catch (e: IOException) {
            throw Exception(e)
        }

    }
}
